src/pages/[productId]/invite/[token].tsx

```diff
@@ -38,17 +40,18 @@ const InvitePage: ProecoNextPage<Props> = ({ team }) => {
   }, [currentUser, team, router, notifySuccessMessage, isValidatingCurrentUser]);
 
   useEffect(() => {
-    if (currentUser && teamUsers.some((teamUser) => teamUser._id === currentUser._id)) {
+    if (currentUser && teamUsers.some((teamUser) => teamUser._id === currentUser._id) && !isApprovingInvite) {
```

Comment:
一度に3つの条件をチェックするのは脳内メモリを逼迫する予感...

```suggestion
    if (!currentUser || isApprovingInvite) {
      return
    }
    if(teamUsers.some((teamUser) => teamUser._id === currentUser._id)) {
      notifySuccessMessage('すでにプロダクトに所属しています');
      router.push(URLS.TEAMS(team.productId));
    }
```
このほうが isApprovingInvite が true の時に some でイテレートしなくて済むので良さそうです